### PR TITLE
12 Add support for ignoreAreas

### LINKED ISSUE #12: Add ignoreAreas into testRun payload
New property is added `ignoreAreas` into request payload and could be processed by server side: https://github.com/Visual-Regression-Tracker/Visual-Regression-Tracker/issues/132
Add ability to pass it as parameter into `track` method to ignore some parts of image

```
class IgnoreArea {
  x: int;
  y: int;
  width: int;
  height: int;
}

```

### PR DESCRIPTION
**Pull request by dcrowe**
Resolves #12

### CODE REVIEW

**Comment by pashidlos** [comment]
@dcrowe I faced an issue when server response has  `diffName` as `None` 
added test case for it, but I'm not able to fix it :(

```
Traceback (most recent call last):
  File "example.py", line 21, in <module>
    vrt.track(TestRun(
  File "/Users/pashidlos/work/vrt/sdk-python/visual_regression_tracker/visualRegressionTracker.py", line 91, in track
    result = self._submitTestResult(test)
  File "/Users/pashidlos/work/vrt/sdk-python/visual_regression_tracker/visualRegressionTracker.py", line 86, in _submitTestResult
    testRunResult = _from_dict(result, TestRunResponse)
  File "/Users/pashidlos/work/vrt/sdk-python/visual_regression_tracker/types.py", line 91, in _from_dict
    obj = dacite.from_dict(clazz, data)
  File "/Users/pashidlos/.conda/envs/sdk-python/lib/python3.8/site-packages/dacite/core.py", line 65, in from_dict
    raise WrongTypeError(field_path=field.name, field_type=field.type, value=value)
dacite.exceptions.WrongTypeError: wrong value type for field "diffName" - should be "str" instead of value "None" of type "NoneType"
```

File: visual_regression_tracker/types.py [hunk 1]
```diff
@@ -69,16 +80,13 @@ def __init__(self, test_run_response: TestRunResponse, api_url: str):
...
+        return dict(kv for kv in key_values if kv[1] is not None)
+
+    data = dataclasses.asdict(obj, dict_factory=dict_factory)
     return data
 
 
 def _from_dict(data, clazz):
-    fields = dataclasses.fields(clazz)
-    obj = clazz(*[data.get(field.name, None) for field in fields])
+    obj = dacite.from_dict(clazz, data)
```
Comment (pashidlos) [inline visual_regression_tracker/types.py:91]: seems like this line is the reason for wrong type casting 
not sure how to let the `str` to accept `None` as well that comes from server

Comment (dcrowe) [reply visual_regression_tracker/types.py:91]: oh yep, i should have expected that. i can fix it.

Comment (dcrowe) [reply visual_regression_tracker/types.py:91]: I've resolved the specific failures of your new tests, but I'm not certain that this covers all user cases. Do you mind taking a look and suggesting any extra fields that need to be optional?

Comment (pashidlos) [reply visual_regression_tracker/types.py:91]: sure, gonna check it!

**Comment by pashidlos** [comment]
works as expected!
@dcrowe thank you for support!

**Comment by pashidlos** [comment]
https://twitter.com/VisualReTracker/status/1326073973742964738

**Pull request merged**